### ISSUE TITLE
Ability to load config from module

### ISSUE THREAD

**Issue by foxx** [bottlepy/bottle#772]
Currently it is not possible to load config from a Python module (similar to the way Django does it).

I'd like to propose the following;

```py
class ConfigDict(dict):
    def load_module(self, module_path):
        """
        Load configuration from module path

        >>> load_module('settings.prod')
        """
        assert isinstance(module_path, str)
        config = importlib.import_module(module_path)
        obj = {key: getattr(config, key) for key in dir(config) if key.isupper()}
        self.update(obj)
```

Happy to put together a tested PR if this proposal is accepted

**Comment by defnull** [comment]
Yes, this is useful.

**Comment by foxx** [comment]
Great stuff, I'll submit a PR by Friday

**Comment by rcarmo** [comment]
Hi there!

May I suggest ensuring that the Python logging configuration becomes part of the overall config? 

I do that [here](https://github.com/rcarmo/ink-bottle/blob/master/lib/config.py) and it's saved me a lot of trouble. An example DictConfig for logging is [here](https://github.com/rcarmo/ink-bottle/blob/master/etc/default.json).

**Comment by defnull** [comment]
The bottle config is flat (i.e. ``keys.with.dots.in.them``) but the logging module expects a nested structure. Some other frameworks need that, too (e.g. beaker). We have ``load_dict`` to flatten a dictionary or json. The reverse would be required to support logging configuration out of bottle. Should not be too hard to implement. I'll open a new request for that.

**Comment by rcarmo** [comment]
Thanks @defnull!

**Comment by foxx** [comment]
So I've added support for this, however I'm unsure if it should use `update()` or `load_dict()`.

Currently, `load_dict()` will squash nested dicts into a single layer dict, by creating a namespace key, where as `update()` won't apply squashing. There are situations where we might not want to squash this config, and we should allow the user to choose.

For example;

```py
>>> import bottle
>>> c = bottle.ConfigDict()
>>> d = dict(a=dict(b=2))
>>> c.update(d)
>>> print(config)
{'a': {'b': 2}}
```

```py
>>> c = bottle.ConfigDict()
>>> d = dict(a=dict(b=2))
>>> c.load_dict(d)
{'a.b': 2}
```

Therefore I have included a `squash` parameter on `load_module()`, if `True` it will use `load_dict()` styling loading, if `False` it will use `update()`.
